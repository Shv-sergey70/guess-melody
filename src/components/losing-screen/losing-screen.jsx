import React, {Fragment} from 'react';
import PropTypes from 'prop-types';
import {connect} from "react-redux";
import {Redirect} from 'react-router-dom';
import {isTimeOver, areAttemptsOver} from '../../reducer/game/selectors';
import PlayAgainLink from "../links/play-again/play-again";
import Route from '../../routes';

const LosingScreen = ({isNoMoreTime, isNoMoreAttempts}) => {
  let content = null;

  switch (true) {
    case isNoMoreTime:
      content = (
        <Fragment>
          <h2 className="result__title">Увы и ах!</h2>
          <p className="result__total result__total--fail">Время вышло! Вы не успели отгадать все мелодии</p>
        </Fragment>
      );

      break;
    case isNoMoreAttempts:
      content = (
        <Fragment>
          <h2 className="result__title">Какая жалость!</h2>
          <p className="result__total result__total--fail">У вас закончились все попытки. Ничего, повезёт в следующий
            раз!</p>
        </Fragment>
      );

      break;
    default:
      content = <Redirect to={Route.MAIN}/>;
  }

  return (
    <section className="result">
      <div className="result__logo">
        <img src="img/melody-logo.png" alt="Угадай мелодию" width="186" height="83"/>
      </div>

      {content}

      <PlayAgainLink />
    </section>
  );
};

LosingScreen.propTypes = {
  isNoMoreTime: PropTypes.bool.isRequired,
  isNoMoreAttempts: PropTypes.bool.isRequired
};

const mapStateToProps = (state) => ({
  isNoMoreTime: isTimeOver(state),
  isNoMoreAttempts: areAttemptsOver(state)

});

export {LosingScreen};
export default connect(mapStateToProps)(LosingScreen);
