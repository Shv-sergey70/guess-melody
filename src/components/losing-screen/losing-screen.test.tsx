import * as React from 'react';
import {configure, shallow} from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import {LosingScreen} from "./losing-screen";

configure({adapter: new Adapter()});

describe(`LosingScreen renders correctly`, () => {
  test(`Losing by time`, () => {
    const tree = shallow(
        <LosingScreen
          isNoMoreTime={true}
          isNoMoreAttempts={false} />
    );

    expect(tree).toMatchSnapshot();
  });

  test(`Losing by attempts`, () => {
    const tree = shallow(
        <LosingScreen
          isNoMoreTime={false}
          isNoMoreAttempts={true} />
    );

    expect(tree).toMatchSnapshot();
  });

  test(`No losing reasons - redirect to main page`, () => {
    const tree = shallow(
        <LosingScreen
          isNoMoreTime={false}
          isNoMoreAttempts={false} />
    );

    expect(tree).toMatchSnapshot();
  });
});
