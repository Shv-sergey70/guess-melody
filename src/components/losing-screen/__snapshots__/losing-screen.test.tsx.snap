// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LosingScreen renders correctly Losing by attempts 1`] = `
<section
  className="result"
>
  <div
    className="result__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="result__title"
  >
    Какая жалость!
  </h2>
  <p
    className="result__total result__total--fail"
  >
    У вас закончились все попытки. Ничего, повезёт в следующий раз!
  </p>
  <Connect(PlayAgain) />
</section>
`;

exports[`LosingScreen renders correctly Losing by time 1`] = `
<section
  className="result"
>
  <div
    className="result__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="result__title"
  >
    Увы и ах!
  </h2>
  <p
    className="result__total result__total--fail"
  >
    Время вышло! Вы не успели отгадать все мелодии
  </p>
  <Connect(PlayAgain) />
</section>
`;

exports[`LosingScreen renders correctly No losing reasons - redirect to main page 1`] = `
<section
  className="result"
>
  <div
    className="result__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <Redirect
    to="/"
  />
  <Connect(PlayAgain) />
</section>
`;
